kyo-combinators/shared/src/main/scala/kyo/Constructors.scala

```diff
@@ -32,112 +56,317 @@ extension (kyoObject: Kyo.type)
 
+    /** Creates an effect from a Try[A] and handles the Try to Abort[Throwable].
+      *
+      * @param _try
+      *   The Try[A] to create an effect from
+      * @return
+      *   An effect that attempts to run the given effect and handles the Try to Abort[Throwable].
+      */
     def fromTry[A, S](_try: => scala.util.Try[A] < S)(using Frame): A < (S & Abort[Throwable]) =
         _try.map(Abort.get(_))
 
+    /** Logs an informational message to the console.
+      *
+      * @param message
+      *   The message to log
+      * @return
+      *   An effect that logs the message to the console
+      */
     inline def logInfo[S](message: => String < S): Unit < (S & IO) =
         message.map(m => Log.info(m))
 
+    /** Logs an informational message to the console with an error.
+      *
+      * @param message
+      *   The message to log
+      * @param err
+      *   The error to log
+      * @return
+      *   An effect that logs the message and error to the console
+      */
     inline def logInfo[S, S1](
         message: => String < S,
         err: => Throwable < S1
     ): Unit < (S & S1 & IO) =
         message.map(m => err.map(e => Log.info(m, e)))
 
+    /** Logs a warning message to the console.
+      *
+      * @param message
+      *   The message to log
+      * @return
+      *   An effect that logs the message to the console
+      */
     inline def logWarn[S](message: => String < S): Unit < (S & IO) =
         message.map(m => Log.warn(m))
 
+    /** Logs a warning message to the console with an error.
+      *
+      * @param message
+      *   The message to log
+      * @param err
+      *   The error to log
+      * @return
+      *   An effect that logs the message and error to the console
+      */
     inline def logWarn[S, S1](
         message: => String < S,
         err: => Throwable < S1
     ): Unit < (S & S1 & IO) =
         message.map(m => err.map(e => Log.warn(m, e)))
 
+    /** Logs a debug message to the console.
+      *
+      * @param message
+      *   The message to log
+      * @return
+      *   An effect that logs the message to the console
+      */
     inline def logDebug[S](message: => String < S): Unit < (S & IO) =
         message.map(m => Log.debug(m))
 
+    /** Logs a debug message to the console with an error.
+      *
+      * @param message
+      *   The message to log
+      * @param err
+      *   The error to log
+      * @return
+      *   An effect that logs the message and error to the console
+      */
     inline def logDebug[S, S1](
         message: => String < S,
         err: => Throwable < S1
     ): Unit < (S & S1 & IO) =
         message.map(m => err.map(e => Log.debug(m, e)))
 
+    /** Logs an error message to the console.
+      *
+      * @param message
+      *   The message to log
+      * @return
+      *   An effect that logs the message to the console
+      */
     inline def logError[S](message: => String < S): Unit < (S & IO) =
```

Comment:
I can't tell what object these are under (extension), but I think they may be confusing names.

If I run

```
val io: A < (Abort[E] & IO) = ???
io.logError(...)
```

I would expect it to tap into the error `E`. Related: we may need to adjust `Log` to handle `Result.Error[E]` or `Throwable`

Reply:
In this case it'd be a companion object method `Kyo.logError(...)`. It seems ok to me. Good point about logging `Result.Error`!